### PR TITLE
Hack6

### PR DESCRIPTION
**Pull request by thesps**
Add fix to dense layers (including large) with binary weights. 
The output of the dense layers was cast to the result type, whereas with binary weights (where -1 is represented with 0) first a transformation of the output is needed.

Also, keras-to-hls.py was removed to simplify development. keras_to_hls.py should be used in its place here. This module is practically the same, but is importable in Python and better for interactive development/debugging.

### CODE REVIEW

**Comment by thesps** [comment]
Now in my simple test the binary dense layer gives the correct output, but the batch normalisation is still a bit wrong.
Keras reference:
Layer3:
6 -10 2 -4 4 -8 0 -6 -2 -4 -12 -4 -4 -4 12 10 -6 -14 -20 18 2 -2 14 -18 16 -18 -6 22 24 4 -6 0
Layer4:
0 0 1 0 0 0 1 0 0 1 0 0 0 0 1 1 0 0 0 1 1 1 1 0 1 0 1 1 1 1 0 0

CSIM Before:
Layer3:
35 27 33 30 34 28 32 29 31 30 26 30 30 30 38 37 29 25 22 41 33 31 39 23 40 23 29 43 44 34 29 32
Layer4:
0 0 1 0 0 1 1 0 1 1 0 1 1 1 1 0 0 1 1 1 0 1 0 1 1 1 1 1 1 0 1 1

CSIM After:
Layer3:
6 -10 2 -4 4 -8 0 -6 -2 -4 -12 -4 -4 -4 12 10 -6 -14 -20 18 2 -2 14 -18 16 -18 -6 22 24 4 -6 0
Layer4:
0 0 1 0 0 0 1 0 1 0 0 0 1 1 1 0 0 1 0 1 0 1 0 1 1 0 1 1 0 0 0 0
